src/main/java/edu/hm/hafner/analysis/parser/ClangAnalyzerPlistParser.java

```diff
@@ -0,0 +1,121 @@
+package edu.hm.hafner.analysis.parser;
+
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.IssueParser;
+import edu.hm.hafner.analysis.ParsingException;
+import edu.hm.hafner.analysis.ReaderFactory;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.util.XmlElementUtil;
+
+
+/**
+ * A parser for the clang-analyzer static analysis warnings.
+ *
+ * @author Andrey Danin
+ */
+public class ClangAnalyzerPlistParser extends IssueParser {
+    private static final long serialVersionUID = 1L;
+    static final String ID = "ClangAnalyzer Plist Parser";
+
+    @Override
+    public boolean accepts(final ReaderFactory readerFactory) {
+        return isXmlFile(readerFactory);
+    }
+
+    @Override
+    public Report parse(final ReaderFactory readerFactory) throws ParsingException {
+        try {
+            Document doc = readerFactory.readDocument();
+
+            XPathFactory xPathFactory = XPathFactory.newInstance();
+            XPath xPath = xPathFactory.newXPath();
+
+            XPathExpression filesPath = xPath.compile("/plist/dict/key[text()='files']/following-sibling::array/string");
+            XPathExpression diagnosticsPath = xPath.compile("/plist/dict/key[text()='diagnostics']/following-sibling::array/dict");
+
+            XPathExpression diagDescriptionPath = compileDiagStrPath(xPath, "description");
+            XPathExpression diagCategoryPath = compileDiagStrPath(xPath, "category");
+            XPathExpression diagTypePath = compileDiagStrPath(xPath, "type");
+            XPathExpression diagLocationLinePath = compileDiagLocationPath(xPath, "line");
+            XPathExpression diagLocationColPath = compileDiagLocationPath(xPath, "col");
+            XPathExpression diagLocationFilePath = compileDiagLocationPath(xPath, "file");
+
+
+            IssueBuilder issueBuilder = new IssueBuilder();
+            Report report = new Report();
+
+            List<String> filesList = new ArrayList<>();
+
+            // Fill files list
+
+            NodeList files = (NodeList)filesPath.evaluate(doc, XPathConstants.NODESET);
+            for (Element filePathStr : XmlElementUtil.nodeListToList(files)) {
+                filesList.add(filePathStr.getTextContent());
+            }
+
+            // Parse errors
+
+            NodeList diagnostics = (NodeList)diagnosticsPath.evaluate(doc, XPathConstants.NODESET);
+            for (Element diag : XmlElementUtil.nodeListToList(diagnostics)) {
+                issueBuilder
+                    .setFileName(filesList.get(extractIntField(diag, diagLocationFilePath)))
```

Comment:
I wonder if it makes sense to move `filesList.get(extractIntField(diag, diagLocationFilePath))` to a method that checks if the index is valid and returns `-` if not. Otherwise a broken file will cause a `RuntimeException`

Reply:
Done